chain/chain/src/store.rs

```diff
@@ -1971,92 +1980,48 @@ impl<'a> ChainStoreUpdate<'a> {
-            self.chain_store
-                .outgoing_receipts
-                .cache_remove(&get_block_shard_id(&block_hash, shard_id));
-            // 2b. Delete incoming receipts (ColIncomingReceipts)
-            store_update.delete(ColIncomingReceipts, &get_block_shard_id(&block_hash, shard_id));
-            self.chain_store
-                .incoming_receipts
-                .cache_remove(&get_block_shard_id(&block_hash, shard_id));
-            // 2c. Delete from chunk_hash_per_height_shard (ColChunkPerHeightShard)
-            store_update.delete(ColChunkPerHeightShard, &get_height_shard_id(height, shard_id));
-            self.chain_store
-                .chunk_hash_per_height_shard
-                .cache_remove(&get_height_shard_id(height, shard_id));
-            // 2d. Delete from next_block_with_new_chunk (ColNextBlockWithNewChunk)
-            store_update
-                .delete(ColNextBlockWithNewChunk, &get_block_shard_id(&block_hash, shard_id));
-            self.chain_store
-                .next_block_with_new_chunk
-                .cache_remove(&get_block_shard_id(&block_hash, shard_id));
-            // 2e. Delete from ColStateHeaders
+            let height_shard_id = get_block_shard_id(&block_hash, shard_id);
+            self.gc_col(ColOutgoingReceipts, &height_shard_id);
+            self.gc_col(ColIncomingReceipts, &height_shard_id);
+            self.gc_col(ColChunkPerHeightShard, &height_shard_id);
+            self.gc_col(ColNextBlockWithNewChunk, &height_shard_id);
             let key = StateHeaderKey(shard_id, block_hash).try_to_vec()?;
-            store_update.delete(ColStateHeaders, &key);
-            // 2f. Delete from ColStateParts
+            self.gc_col(ColStateHeaders, &key);
             // Already done, check chain.clear_downloaded_parts()
         }
 
         // 3. Delete block_hash-indexed data
-        let block_hash_ref = block_hash.as_ref();
-        // 3a. Delete block (ColBlock)
-        store_update.delete(ColBlock, block_hash_ref);
-        self.chain_store.blocks.cache_remove(&block_hash.into());
-        // 3b. Delete block header (ColBlockHeader) - don't do because header sync needs headers
-        // 3c. Delete block extras (ColBlockExtra)
-        store_update.delete(ColBlockExtra, block_hash_ref);
-        self.chain_store.block_extras.cache_remove(&block_hash.into());
-        // 3d. Delete from next_block_hashes (ColNextBlockHashes)
-        store_update.delete(ColNextBlockHashes, block_hash_ref);
-        self.chain_store.next_block_hashes.cache_remove(&block_hash.into());
-        // 3e. Delete from ColChallengedBlocks
-        store_update.delete(ColChallengedBlocks, block_hash_ref);
-        // 3f. Delete from ColBlocksToCatchup
-        store_update.delete(ColBlocksToCatchup, block_hash_ref);
-        // 3g. Delete from KV state changes
+        // Don't delete block header (ColBlockHeader) - because header sync needs headers
```

Comment:
All comments about why we don't delete something should be in `SHOULD_COL_GC` comments.